ydb/library/yql/core/common_opt/yql_co_simple1.cpp

```diff
@@ -70,6 +70,42 @@ TExprNode::TPtr ExpandPgNot(const TExprNode::TPtr& input, TExprContext& ctx) {
         .Build();
 }
 
+template <bool Bool>
+TExprNode::TPtr ExpandPgIsTorF(const TExprNode::TPtr& input, TExprContext& ctx) {
+    return ctx.Builder(input->Pos())
+        .Callable("ToPg")
+            .Callable(0, "Coalesce")
+                .Callable(0, "FromPg")
+                    .Callable(0, "PgOp")
+                        .Atom(0, "=")
+                        .Add(1, input->ChildPtr(0))
+                        .Add(2, MakePgBool<Bool>(input->Pos(), ctx))
+                    .Seal()
+                .Seal()
+            .Add(1, MakeBool<false>(input->Pos(), ctx))
+            .Seal()
+        .Seal()
+        .Build();
+}
+
+TExprNode::TPtr ExpandPgIsTrue(const TExprNode::TPtr& input, TExprContext& ctx) {
+    return ExpandPgIsTorF<true>(input, ctx);
+}
+
+TExprNode::TPtr ExpandPgIsFalse(const TExprNode::TPtr& input, TExprContext& ctx) {
+    return ExpandPgIsTorF<false>(input, ctx);
+}
+
+TExprNode::TPtr ExpandPgIsUnknown(const TExprNode::TPtr& input, TExprContext& ctx) {
```

Comment:
is unknown = is null = NOT exists

Reply:
Tests do pass as coded now

Reply:
Fixed. Code locality improved but expr tree got larger for no reason.